hnn_core/network.py

```diff
@@ -832,6 +903,320 @@ def _get_src_type_and_pos(self, gid):
+        nc_dict['lamtha'] = 3.
+        receptor = 'ampa'
+        for loc in ['proximal', 'distal']:
+            key = f'gbar_L2Pyr_{target_cell}'
+            nc_dict['A_weight'] = self.params[key]
+            connectivity = self._all_to_all_connect(
+                src_types, src_gids, target_types, target_gids,
+                loc, receptor, nc_dict)
+            self.connectivity_list.extend(connectivity)
+
+        # layer2 Basket -> layer5 Pyr
+        src_cell = 'L2Basket'
+        src_gids = self.gid_ranges[_long_name(src_cell)]
+        src_types = np.repeat(src_cell, len(src_gids))
+        nc_dict['lamtha'] = 50.
+        key = f'gbar_L2Basket_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'distal'
+        receptor = 'gabaa'
+        connectivity = self._all_to_all_connect(
+            src_types, src_gids, target_types, target_gids,
+            loc, receptor, nc_dict)
+        self.connectivity_list.extend(connectivity)
+
+        # xx -> layer2 Basket
+        src_cell = 'L2Pyr'
+        src_gids = self.gid_ranges[_long_name(src_cell)]
+        src_types = np.repeat(src_cell, len(src_gids))
+        target_cell = 'L2Basket'
+        target_gids = self.gid_ranges[_long_name(target_cell)]
+        target_types = np.repeat(target_cell, len(target_gids))
+        nc_dict['lamtha'] = 3.
+        key = f'gbar_L2Pyr_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'soma'
+        receptor = 'ampa'
+        connectivity = self._all_to_all_connect(
+            src_types, src_gids, target_types, target_gids,
+            loc, receptor, nc_dict)
+        self.connectivity_list.extend(connectivity)
+
+        src_cell = 'L2Basket'
+        src_gids = self.gid_ranges[_long_name(src_cell)]
+        src_types = np.repeat(src_cell, len(src_gids))
+        nc_dict['lamtha'] = 20.
+        key = f'gbar_L2Basket_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        loc = 'soma'
+        receptor = 'gabaa'
+        connectivity = self._all_to_all_connect(
+            src_types, src_gids, target_types, target_gids,
+            loc, receptor, nc_dict)
+        self.connectivity_list.extend(connectivity)
+
+        # xx -> layer5 Basket
+        src_cell = 'L5Basket'
+        src_gids = self.gid_ranges[_long_name(src_cell)]
+        src_types = np.repeat(src_cell, len(src_gids))
+        target_cell = 'L5Basket'
+        target_gids = self.gid_ranges[_long_name(target_cell)]
+        target_types = np.repeat(target_cell, len(target_gids))
+        nc_dict['lamtha'] = 20.
+        loc = 'soma'
+        receptor = 'gabaa'
+        key = f'gbar_L5Basket_{target_cell}'
+        nc_dict['A_weight'] = self.params[key]
+        connectivity = self._all_to_all_connect(
+            src_types, src_gids, target_types, target_gids,
+            loc, receptor, nc_dict, allow_autapses=False)
+        self.connectivity_list.extend(connectivity)
```

Comment:
I feel the readability has taken a serious hit here. It used to be 5 lines. Now it has doubled. How can we keep it readable while extending the functionality?

Reply:
okay I pushed a commit for this ...

Reply:
Thanks for moving that code!! Definitely cleans it up a lot.